src/prefect/tasks.py

```diff
@@ -381,6 +379,12 @@ def __init__(
         self.cache_expiration = cache_expiration
         self.refresh_cache = refresh_cache
 
+        if not persist_result:
+            self.cache_policy = NONE
+            if cache_policy:
```

Comment:
This may need to be more explicit (a user could set the cache policy to NONE themselves for example), or it could still be NotSet

Reply:
Good call! Added in https://github.com/PrefectHQ/prefect/pull/14102/commits/03d390b12006b9c3ac1225a61dad1ad3ccf204e5